### PR TITLE
nick11967 judge

### PR DESCRIPTION
**Pull request by nick11967**
10127 Ones
10157 Expressions
846 Steps

### CODE REVIEW

File: nick11967/Ch 6/48 Steps/steps.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+vector<int> max_len;
+
+int step(int x)
+{
+    int steps;
```
Comment (nick11967) [inline nick11967/Ch 6/48 Steps/steps.cpp]: remove 'steps'

File: nick11967/Ch 6/48 Steps/steps.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        // printf("#%d: %d\n", i, max_len[i]);
+        if (x <= max_len[i])
+        {
+            return i;
+        }
+    }
+    while (1)
+    {
+        max_len.push_back(max_len[size - 1] + (int)floor((double)size / 2 + 0.5));
```
Comment (lucetre) [inline nick11967/Ch 6/48 Steps/steps.cpp]: `(int)floor((double)size / 2 + 0.5)`
into
`(size+1)/2`

File: nick11967/Ch 6/48 Steps/steps.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
+// UVa ID: 846
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+vector<int> max_len;
```
Comment (nick11967) [inline nick11967/Ch 6/48 Steps/steps.cpp]: use unsigned int

File: nick11967/Ch 6/48 Steps/steps.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+int step(int x)
+{
+    int steps;
+    int size = max_len.size();
+    for (int i = 1; i < size; i++)
+    {
+        // printf("#%d: %d\n", i, max_len[i]);
+        if (x <= max_len[i])
```
Comment (nick11967) [inline nick11967/Ch 6/48 Steps/steps.cpp:16]: max_len[i] can be bigger than integer

Comment (lucetre) [reply nick11967/Ch 6/48 Steps/steps.cpp:16]: sgtm

File: nick11967/Ch 8/57 Little Bishops/littlebishops.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    return true;
+}
+
+long long bishop(int x, int y)
+{
+    if (x == n)
+        return 0;
+    if (num == k)
+        return 1;
+    long long ans = 0;
```
Comment (lucetre) [inline nick11967/Ch 8/57 Little Bishops/littlebishops.cpp:33]: add base condition to reduce execution time

File: nick11967/Ch 8/57 Little Bishops/littlebishops.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+using namespace std;
+
+int n, k, num = 0;
+// num: how many bishops on the board until now
+
+int pos[MAX_BIS + 1][2]; // record pos of bishops
+
+bool possible(int x, int y)
+{
+    for (int i = 0; i < num; i++)
```
Comment (lucetre) [inline nick11967/Ch 8/57 Little Bishops/littlebishops.cpp:16]: consider using check array

File: nick11967/Ch 6/44 Expressions/expressions2.cpp [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+    for (i = 0; i <= 150; i++)
+    {
+        for (j = 0; j <= 150; j++)
+        {
+            if (i == 0 && j == 0)
+                f[i][j] = 1;
+            else if (i <= j)
+                f[i][j] = e[i];
```
Comment (lucetre) [inline nick11967/Ch 6/44 Expressions/expressions2.cpp:35]:
```
else if (i < j)
  f[i][j] = f[i][i]
```

Comment (lucetre) [reply nick11967/Ch 6/44 Expressions/expressions2.cpp:35]: sgtm

**Review by lucetre — APPROVED** [review]

**Pull request merged**